sway/input/seatop_down.c

```diff
@@ -72,15 +73,29 @@ static void handle_tablet_tool_motion(struct sway_seat *seat,
 	}
 }
 
+static void handle_destroy(struct wl_listener *listener, void *data) {
+	struct seatop_down_event *e =
+		wl_container_of(listener, e, surface_destroy);
+	if (e) {
+		seatop_begin_default(e->seat);
+	}
+}
+
 static void handle_unref(struct sway_seat *seat, struct sway_container *con) {
 	struct seatop_down_event *e = seat->seatop_data;
 	if (e->con == con) {
 		seatop_begin_default(seat);
 	}
 }
 
+static void handle_end(struct sway_seat *seat) {
+	struct seatop_down_event *e = seat->seatop_data;
+	wl_list_remove(&e->surface_destroy.link);
+}
+
 static const struct sway_seatop_impl seatop_impl = {
 	.button = handle_button,
+	.end = handle_end,
```

Comment:
Nit: put this next to `.unref`, not in the middle of the pointer handlers.

Reply:
Ah alright, though it was alphabetical ^^